internal/receiver/databricksreceiver/paginator.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databricksreceiver
+
+import "fmt"
+
+// paginator is extracted from apiPaginator for swapping out in unit tests
+type paginator interface {
+	jobsList() (out []job, err error)
+	activeJobRuns() (out []jobRun, err error)
+	completedJobRuns(jobID int, time int64) (out []jobRun, err error)
+}
+
+// apiPaginator handles pagination (responses specify hasMore=true/false) and
+// combines the returned objects into one array.
+type apiPaginator struct {
+	unmarshaller unmarshaller
+	limit        int
+}
+
+func newPaginator(api databricksAPI) apiPaginator {
+	return apiPaginator{
+		unmarshaller: unmarshaller{api: api},
+		limit:        25, // max is 25
```

Comment:
Do you imaging this ever needing to change and being worth a config option?

Reply:
I was thinking it wouldn't since the max is only 25, but I've made it configurable anyway since it does help with testing.